corpus/segments.py

```diff
@@ -261,6 +262,33 @@ def run(self):
                 f.writelines(lines[start:end])
 
 
+class DynamicSplitSegmentFileJob(Job):
+
+    def __init__(self, segment_file, concurrent=1):
+        self.segment_file = segment_file
+        self.concurrent = concurrent
+        self.segment_dir = self.output_path("split", directory=True)
+
+    def tasks(self):
+        yield Task("run", resume="run", mini_task=True)
+
+    def run(self):
+        with open(tk.uncached_path(self.segment_file), "rt") as f:
+            lines = [l for l in f.readlines() if len(l.strip()) > 0]
+
+        nb_seg = len(lines)
+        self.concurrent = self.concurrent.get()
```

Comment:
this will fail when concurrent keeps it's default value of 1, please do a check for Delayed here